utils/components.py

```diff
@@ -298,3 +307,359 @@ def ilab_importer_op(repository: str, release: str, base_model: dsl.Output[dsl.M
             f"ilab --config=DEFAULT model download --repository {repository} --release {release} --model-dir {base_model.path}"
         ],
     )
+
+
+@dsl.component(base_image=RUNTIME_GENERIC_IMAGE)
+def test_model_connection(secret_name: str):
+    import base64
+    import json
+    import os
+    import sys
+    import time
+
+    import requests
+    from kubernetes import client, config
+    from kubernetes.client.rest import ApiException
+
+    config.load_incluster_config()
+
+    model_endpoint = ""
+    model_name = ""
+    model_api_key = ""
+    with open(
+        "/var/run/secrets/kubernetes.io/serviceaccount/namespace", "r"
+    ) as namespace_path:
+        namespace = namespace_path.readline()
+
+    with client.ApiClient() as api_client:
+        core_api = client.CoreV1Api(api_client)
+
+        try:
+            secret = core_api.read_namespaced_secret(secret_name, namespace)
+            print(f"Reading secret {secret_name} data...")
+            model_api_key = base64.b64decode(secret.data["api_key"]).decode("utf-8")
+            model_name = base64.b64decode(secret.data["model_name"]).decode("utf-8")
+            model_endpoint = base64.b64decode(secret.data["endpoint"]).decode("utf-8")
+        except (ApiException, KeyError) as e:
+            print(f"""
+            ####################################################### ERROR ################################################################
+            # Error reading {secret_name}. Ensure you created a secret with this name in namespace {namespace} and has 'api_key' present #
```

Comment:
The error message is a bit misleading because it also checks for `model_name` and `endpoint` and the error message doesn't mention those.